### PR TITLE
bugfix: openapi code generator to consider schema changes

### PR DESCRIPTION
**Pull request by DeepikaSridhar2112**
Issue description : OpenApi code generator fails to regenerate the classes when changes are made to schema specifications.

Analysis: Open Api code generator task, org.openapitools.generator.gradle.plugin.tasks.GenerateTask is present in openapi-generator-gradle-plugin-5.4.0.jar. It scans changes across parameters provided as inputs and determines if code has to be regenerated on invocation.If schema changes made directly in openapi.yml it detects the change and re-generates the code. However it isnt aware of the location of the schema files refered via reference($ref) tags. Hence the open api code generator doesn't detect changes made to schema files.

Resolution: Adding schema location as an parameter to GenerateTask.kt resolves the issue. The plugin when executed detects changes made to reference schema files

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by DeepikaSridhar2112** [comment]
> Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.
> 
> Let me know if you need help fixing it.
> 
> Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

Hi,
Updated the commits to get it linked to github account.Can you please check

**Comment by wing328** [comment]
thanks for the PR. would it be possible to add a test to cover the issue?

File: modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/GenerateTask.kt [hunk 1]
```diff
@@ -80,6 +80,15 @@ open class GenerateTask @Inject constructor(private val objectFactory: ObjectFac
     @Input
     val generatorName = project.objects.property<String>()
 
+    /**
+     * This is the configuration for reference paths where schemas for openapi generation are stored
+     * The schema target directory into which code will be generated
```
Comment (codefish1) [inline modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/tasks/GenerateTask.kt:85]: Is this not the location from which we generate schema?

```suggestion
     * The directory which contains the additional schema files
```

**Comment by codefish1** [comment]
> thanks for the PR. would it be possible to add a test to cover the issue?
> 
> tests can be found in modules/openapi-generator-gradle-plugin/src/test/kotlin/

@wing328 We have been running into problems where we've split a large OpenAPI specification into numerous files, with a file for each object.  These are all in various sub folders and pulled in using `$red: "/.schemas/foo.yml` for example.  Without marking the schema folder as an input directory for the generate task, Gradle will use it's cache since it doesn't think any inputs have changed, despite you making changes to `foo.yml`.

How would you suggest we add a test for this?  I assume you already have tests covering how you import schemas via reference, and this change is around the Gradle caching when a change has been made after your first build.

**Comment by codefish1** [comment]
@wing328 what would you suggest we do for a test on this as requested?

**Comment by wing328** [comment]
all tests passed via https://github.com/OpenAPITools/openapi-generator/pull/19701

**Pull request merged**